### DISCUSSION TITLE
Livewire, Blade Component, Alpine and @entangle

### DISCUSSION THREAD

**Discussion by vinnydude** [livewire/livewire#5988] [1. Help] [answered]
I'm just trying to clean up a bit of code and I've hit a stumbling block.

I have a livewire form, and I've created reusable blade components for each form element that I use.

Live wire should be taking care of the values, however, some of my components need to use alpine, and some of those need to reference the value of the form field.

My issue is, I'm trying to use the @entangle method and pass in the name of the form field dynamically. This doesn't seem to be working at all though, it's referencing the livewire component and not the variable I'm passing in.

Does anyone have any idea how I pass a variable into @entangle?

**Comment by vinnydude** [comment, marked as answer]
I've got it figured out.

I have my alpine broken out at the bottom and I was tripping myself up trying to entangle it there using @entangle.

It SHOULD be in my x-data getting passed to my alpine script like so:

`x-data="select($wire.entangle('{{ $name }}'))"`

**Comment by hussain-alkhatib** [comment]
sory i have problem with that
in v2 when user reload page it was take value from local storage and send it to component
but in v3 it get default value frome component and set it local storage

i already update my code to :

<div class="col-12 col-md-6 col-lg-3 mb-2 mb-lg-0"
        x-data="{
            search: $persist(@entangle('search').live),
        }" >
        <input type="search" placeholder="search" class="form-control" wire:model.live="search">
</div>
so how can i fix that
